src/appsignal/config.py

```diff
@@ -73,19 +73,26 @@ class Config:
         list[DefaultInstrumentation], list(get_args(DefaultInstrumentation))
     )
 
-    def __init__(self, options: Optional[Options]):
-        self.initial_source = options
-        self.system_source = self.load_from_system()
-        self.environment_source = self.load_from_environment()
-        self.options = self.DEFAULT_CONFIG | self.system_source
-        if self.initial_source:
-            self.options = self.options | self.initial_source
-        self.options = self.options | self.environment_source
-
-    def load_from_system(self) -> Options:
+    def __init__(self, options: Optional[Options] = None):
+        self.sources = dict(
```

Comment:
Should I extract this as a new TypeDict? so it's easier to add type hints?

Reply:
I like the idea of having a Sources typed dict, but if you run into too much type casting pain, feel free to drop it.

Either way, could you could add a type hint to this `Config` class about the `sources` attribute? Either `sources: dict[string, Options]` or with the specific typed dict if you go that way.